build-system/tasks/lint.js

```diff
@@ -132,14 +133,48 @@ function runLinter(path, stream, options) {
  *
  * @return {!Array<string>}
  */
-function filesInPr() {
+function jsFilesInPr() {
   const filesInPr =
         getStdout('git diff --name-only master...HEAD').trim().split('\n');
   return filesInPr.filter(function(file) {
     return path.extname(file) == '.js';
   });
 }
 
+/**
+ * Checks if there are .eslintrc changes in this PR, in which case we must lint
+ * all files.
+ *
+ * @return {boolean}
+ */
+function eslintrcChangesInPr() {
+  const filesInPr =
+        getStdout('git diff --name-only master...HEAD').trim().split('\n');
+  return filesInPr.filter(function(file) {
+    return path.basename(file).includes('.eslintrc');
+  }).length > 0;
+}
+
+/**
+ * Sets the list of files to be linted.
+ *
+ * @param {!Array<string>} files
+ */
+function setFiles(files) {
+  config.lintGlobs =
+      config.lintGlobs.filter(e => e !== '**/*.js').concat(files);
```

Comment:
Do you want to check paths that end with `**/*.js` instead of strictly matching?

Reply:
This works because there's only one positive glob in `config.lintGlobs`:

https://github.com/ampproject/amphtml/blob/5088f70bb23b2db6eff7da99504f0f02c9b98d7d/build-system/config.js#L119-L134